web_src/js/features/repo-legacy.js

```diff
@@ -29,6 +29,24 @@ import {hideElem, showElem} from '../utils/dom.js';
 
 const {csrfToken} = window.config;
 
+// if there are draft comments (more than 20 chars), confirm before reloading, to avoid losing comments
+function reloadConfirmDraftComment() {
+  const commentTextareas = [
+    document.querySelector('.edit-content-zone:not(.gt-hidden) textarea'),
+    document.querySelector('.edit_area'),
+  ];
+  for (const textarea of commentTextareas) {
+    if (textarea && textarea.value.trim().length > 20) {
```

Comment:
Why it's 20 not other number?

Reply:
Just a magic number.

I think most users won't feel sad if they lose a comment with 10 or 20 chars

But if they have typed 50 chars and the comment is lost, they will be very unhappy.

Reply:
Added more comments in https://github.com/go-gitea/gitea/pull/23014/commits/afef2c4ea9fedca0d35db8ee1aa4c1f0a8b54419